openmc/search.py

```diff
@@ -0,0 +1,249 @@
+        check_type('model_builder return', model, openmc.model.Model)
+        self._model_builder = model_builder
+
+    @property
+    def initial_guess(self):
+        return self._initial_guess
+
+    @initial_guess.setter
+    def initial_guess(self, initial_guess):
+        if initial_guess is not None:
+            check_type('initial_guess', initial_guess, Real)
+        self._initial_guess = initial_guess
+
+    @property
+    def bracket(self):
+        return self._bracket
+
+    @bracket.setter
+    def bracket(self, bracket):
+        if bracket is not None:
+            check_iterable_type('bracket', bracket, Real)
+            check_length('bracket', bracket, 2)
+        self._bracket = bracket
+
+    @property
+    def target_keff(self):
+        return self._target_keff
+
+    @target_keff.setter
+    def target_keff(self, target_keff):
+        check_type('target_keff', target_keff, Real)
+        self._target_keff = target_keff
+
+    @property
+    def print_iterations(self):
+        return self._print_iterations
+
+    @print_iterations.setter
+    def print_iterations(self, print_iterations):
+        check_type('print_iterations', print_iterations, bool)
+        self._print_iterations = print_iterations
+
+    @property
+    def print_output(self):
+        return self._print_output
+
+    @print_output.setter
+    def print_output(self, print_output):
+        check_type('print_output', print_output, bool)
+        self._print_output = print_output
+
+    @property
+    def model_args(self):
+        return self._model_args
+
+    @model_args.setter
+    def model_args(self, model_args):
+        check_type('model_args', model_args, dict)
+        self._model_args = model_args
+
+    def _search_function(self, guess):
+        # Build the model
+        model = self.model_builder(guess, **self.model_args)
+
+        # Run the model
+        keff = model.execute(output=self._print_output)
+
+        # Close the model to ensure HDF5 will allow access during the next
+        # OpenMC execution
+        model.close()
+
+        # Record the history
+        self.guesses.append(guess)
+        self.keffs.append(keff[0])
+        self.keff_uncs.append(keff[1])
+
+        if self._print_iterations:
+            text = 'Iteration: {}; Guess of {:.2E} produced a keff of ' + \
+                '{:1.5f} +/- {:1.5f}'
+            print(text.format(self._i, guess, keff[0], keff[1]))
+        self._i += 1
+
+        return (keff[0] - self.target_keff)
+
+    def search(self, tol=None, bracketed_method='brentq', **kwargs):
+        """Searches for the target eigenvalue with the Newton-Raphson method
+
+        Parameters
+        ----------
+        tol : Real
+            Tolerance to pass to the search method
+        bracketed_method : {'brentq', 'brenth', 'ridder', 'bisect'}, optional
+            Solution method to use; only applies if
+            :param:`bracket` is set, otherwise the Newton method is used.
```

Comment:
100% right.  I have run it and the first guess is basically a waste of time.  The cases I've run with the newton (and as you say, really just Secant) method did eventually converge but took way longer than one would want.  I tried writing this in a way to provide flexibility and not to guarantee the 'best' way to answer each individual problem.